src/Beacon.jsx

```diff
@@ -163,23 +164,21 @@ export class Beacon extends Component {
       if (boundsChanged || (this.state.targetElement !== targetElement)) {
         this.setState({ targetElement, targetBounds }); // eslint-disable-line react/no-did-update-set-state
       }
-    }
 
-    if (this.state.tooltip && !this.state.arrowPosition) {
-      // Calculate relative arrow position when ref is available
-      setTimeout(() => {
```

Comment:
I need proof that this is needed. Everything seems to work fine without it as far as I can see.